### PR TITLE
helm gpu-operator instead of ClusterResourceSet

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind failing-test
/kind flake

**What this PR does / why we need it**:

While debugging GPU E2E test flakes, I observed the gpu-operator version we were using was rather old. Furthermore, we are using a static implementation via ClusterResourceSet.

This PR replaces the static ClusterResourceSet implementation with a helm install implementation using the official nvidia gpu-operator chart @ HEAD (i.e., we will always pull in the latest), following the successful pattern established for cloud-provider-azure and calico.

Documentation updated and CRS files and scripts cleaned up.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: config/default/manager_image_patch.yaml [hunk 1]
```diff
@@ -8,5 +8,5 @@ spec:
     spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: gcr.io/k8s-staging-cluster-api-azure/cluster-api-azure-controller:latest
+      - image: localhost:5000/ci-e2e/cluster-api-azure-controller-amd64:20230125225333
```
Comment (CecileRobertMichon) [inline config/default/manager_image_patch.yaml:11]: accidental add?

File: config/default/manager_pull_policy.yaml [hunk 2]
```diff
@@ -8,4 +8,4 @@ spec:
     spec:
       containers:
       - name: manager
-        imagePullPolicy: Always
+        imagePullPolicy: IfNotPresent
```
Comment (CecileRobertMichon) [inline config/default/manager_pull_policy.yaml:11]: ditto

File: docs/book/src/topics/gpu.md [hunk 3]
```diff
@@ -39,31 +39,17 @@ Apply the manifest from the previous step to your management cluster to have CAP
...
 machinedeployment.cluster.x-k8s.io/azure-gpu-md-0 serverside-applied
 azuremachinetemplate.infrastructure.cluster.x-k8s.io/azure-gpu-md-0 serverside-applied
 kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io/azure-gpu-md-0 serverside-applied
-clusterresourceset.addons.cluster.x-k8s.io/crs-gpu-operator serverside-applied
-configmap/nvidia-clusterpolicy-crd serverside-applied
-configmap/nvidia-gpu-operator-components serverside-applied
-clusterresourceset.addons.cluster.x-k8s.io/azure-gpu-crs-0 serverside-applied
 ```
 
-<aside class="note">
```
Comment (CecileRobertMichon) [inline docs/book/src/topics/gpu.md:56]: ❤️

File: config/default/manager_pull_policy.yaml (hunk 2 above)
Comment (mboersma) [inline config/default/manager_pull_policy.yaml:11]: Same here, mistakenly committed.

File: config/default/manager_image_patch.yaml (hunk 1 above)
Comment (jackfrancis) [reply config/default/manager_image_patch.yaml:11]: ugh!

**Comment by jackfrancis** [comment]
/retest

(transient broken link short-circuited the lint job)

File: test/e2e/gpu_operator.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	clusterProxy := input.BootstrapClusterProxy.GetWorkloadCluster(ctx, input.Namespace.Name, input.ClusterName)
+	InstallNvidiaGPUOperatorChart(ctx, clusterProxy)
+}
+
+// InstallNvidiaGPUOperatorChart installs the official nvidia/gpu-operator helm chart
+// and validates that expected pods exist and are Ready.
+func InstallNvidiaGPUOperatorChart(ctx context.Context, clusterProxy framework.ClusterProxy) {
+	By("Installing nvidia/gpu-operator via helm")
+	values := &helmVals.Options{}
+	InstallHelmChart(ctx, clusterProxy, nvidiaGPUOperatorNamespace, nvidiaHelmChartRepoURL, nvidiaGPUOperatorHelmChartName, nvidiaGPUOperatorHelmReleaseName, values)
```
Comment (jackfrancis) [inline test/e2e/gpu_operator.go:68]: I opted to not worry about identifying and waiting for specific daemonsets/pods like we do for cloud-provider-azure, as the specific deliverables of the nvidia gpu-operator helm chart aren't something we're able to predictively identify (IME) from version to version.

It seems like retry tolerance we have for identifying a `nvidia.com/gpu`-allocatable node is sufficient for this purpose, and that's the key outcome of the gpu-operator. And then of course we also run a CUDA test pod. (though I observed that it seems the gpu-operator helm chart actually does this itself now as a self-bootstrap test, so maybe we don't need to maintain that separate step anymore and we can trust that the application of the expected 'allocatable' metadata is a sufficiently strong signal of success?)

**Comment by jackfrancis** [comment]
/retest

failure in OOT scenario:

```
Timed out waiting for 2 nodes to be created for MachineDeployment capz-e2e-5a6ef3/capz-e2e-5a6ef3-oot-md-0
Expected
    <int>: 0
to equal
    <int>: 2 failed [FAILED] Timed out after 1500.000s.
Timed out waiting for 2 nodes to be created for MachineDeployment capz-e2e-5a6ef3/capz-e2e-5a6ef3-oot-md-0
Expected
    <int>: 0
to equal
    <int>: 2
```

File: test/e2e/gpu_operator.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+type GPUOperatorSpecInput struct {
+	BootstrapClusterProxy framework.ClusterProxy
+	Namespace             *corev1.Namespace
+	ClusterName           string
+}
+
+// InstallGPUOperator installs the official nvidia/gpu-operator helm chart.
+func InstallGPUOperator(ctx context.Context, inputGetter func() GPUOperatorSpecInput) {
+	var (
+		specName = "daemonsets"
```
Comment (CecileRobertMichon) [inline test/e2e/gpu_operator.go:49]: wrong spec name?

File: test/e2e/gpu_operator.go (hunk 4 above)
Comment (CecileRobertMichon) [reply test/e2e/gpu_operator.go:68]: >  validates that expected pods exist and are Ready.

Fair, let's remove this part from the comment?

File: test/e2e/helpers.go [hunk 6]
```diff
@@ -75,6 +75,7 @@ const (
 	retryableOperationTimeout             = 30 * time.Second
 	retryableDeleteOperationTimeout       = 3 * time.Minute
 	retryableOperationSleepBetweenRetries = 3 * time.Second
+	helmInstallTimeout                    = 3 * time.Minute
```
Comment (CecileRobertMichon) [inline test/e2e/helpers.go:78]: 👍

**Review by CecileRobertMichon — COMMENTED** [review]
lgtm aside from few nits

File: test/e2e/gpu_operator.go (hunk 4 above)
Comment (jackfrancis) [reply test/e2e/gpu_operator.go:67]: thx, fixed!

File: test/e2e/gpu_operator.go (hunk 5 above)
Comment (jackfrancis) [reply test/e2e/gpu_operator.go]: thx!

File: docs/book/src/topics/gpu.md [hunk 7]
```diff
@@ -75,38 +61,58 @@ azure-gpu-control-plane-t94nm    azure:////subscriptions/<subscription_id>/resou
 azure-gpu-md-0-f6b88dd78-vmkph   azure:////subscriptions/<subscription_id>/resourceGroups/azure-gpu/providers/Microsoft.Compute/virtualMachines/azure-gpu-md-0-gcc8v            Running   v1.22.1
 ```
 
-Install a [CNI](https://cluster-api.sigs.k8s.io/user/quick-start.html#deploy-a-cni-solution) of your choice.
-Once the nodes are `Ready`, run the following commands against the workload cluster to check if all the `gpu-operator` resources are installed:
+... and then you can install a [CNI](https://cluster-api.sigs.k8s.io/user/quick-start.html#deploy-a-cni-solution) of your choice.
+
+Once all nodes are `Ready`, you can install the official NVIDIA gpu-operator via helm.
```
Comment (mboersma) [inline docs/book/src/topics/gpu.md:66]:
```suggestion
Once all nodes are `Ready`, install the official NVIDIA gpu-operator via Helm.
```

File: docs/book/src/topics/gpu.md [hunk 8]
```diff
@@ -75,38 +61,58 @@ azure-gpu-control-plane-t94nm    azure:////subscriptions/<subscription_id>/resou
...
+
+### Install nvidia gpu-operator helm chart
```
Comment (mboersma) [inline docs/book/src/topics/gpu.md:68]:
```suggestion
### Install nvidia gpu-operator Helm chart
```

File: docs/book/src/topics/gpu.md [hunk 9]
```diff
@@ -75,38 +61,58 @@ azure-gpu-control-plane-t94nm    azure:////subscriptions/<subscription_id>/resou
...
+
+We assume you have helm. If not, official installation instructions for your environment can be found [here](https://helm.sh).
```
Comment (mboersma) [inline docs/book/src/topics/gpu.md:70]:
```suggestion
If you don't have `helm`, installation instructions for your environment can be found [here](https://helm.sh).
```

File: docs/book/src/topics/gpu.md [hunk 10]
```diff
@@ -75,38 +61,58 @@ azure-gpu-control-plane-t94nm    azure:////subscriptions/<subscription_id>/resou
...
+
+First, grab the kubeconfig from your newly created cluster and save it to a file:
+
+```bash
+$ clusterctl get kubeconfig azure-gpu > ./azure-gpu-cluster.conf
+```
+
+Now we can use helm to install the official chart:
```
Comment (mboersma) [inline docs/book/src/topics/gpu.md:78]:
```suggestion
Now we can use Helm to install the official chart:
```

File: docs/book/src/topics/gpu.md [hunk 11]
```diff
@@ -75,38 +61,58 @@ azure-gpu-control-plane-t94nm    azure:////subscriptions/<subscription_id>/resou
...
-gpu-feature-discovery-66d4f                1/1     Running     0          2s
-nvidia-container-toolkit-daemonset-lxpkx   1/1     Running     0          3m11s
-nvidia-dcgm-exporter-wwnsw                 1/1     Running     0          5s
-nvidia-device-plugin-daemonset-lpdwz       1/1     Running     0          13s
-nvidia-device-plugin-validation            0/1     Completed   0          10s
-nvidia-driver-daemonset-w6lpb              1/1     Running     0          3m16s
+$ helm install --kubeconfig ./azure-gpu-cluster.conf --repo https://helm.ngc.nvidia.com/nvidia gpu-operator --generate-name
 ```
 
+The installation of GPU drivers via gpu-operator will take several minutes. Coffee or tea may be appropriate at this time.
```
Comment (mboersma) [inline docs/book/src/topics/gpu.md:84]: ☕ ❤️

**Review by mboersma — COMMENTED** [review]
lgtm, just had some small doc suggestions.

File: docs/book/src/topics/gpu.md (hunk 7 above)
Comment (jackfrancis) [reply docs/book/src/topics/gpu.md]: thanks!

File: docs/book/src/topics/gpu.md (hunk 8 above)
Comment (jackfrancis) [reply docs/book/src/topics/gpu.md]: danke!

File: docs/book/src/topics/gpu.md (hunk 9 above)
Comment (jackfrancis) [reply docs/book/src/topics/gpu.md]: merci!

File: docs/book/src/topics/gpu.md (hunk 10 above)
Comment (jackfrancis) [reply docs/book/src/topics/gpu.md]: tack!

**Review by mboersma — COMMENTED** [review]
/lgtm

Thanks for doing this! It's a big chunk of work and the old way was increasingly rusty.

**Pull request merged**